cpp/oneapi/dal/algo/knn/BUILD

```diff
@@ -22,7 +22,7 @@ dal_test_suite(
         "test/*.cpp",
     ]),
     dal_deps = [
-        "@onedal//cpp/oneapi/dal/test/engine/metrics",
+        "@onedal//cpp/oneapi/dal/test/engine/metrics:metrics",
```

Comment:
The change is equivalent to the old line

Reply:
To fix the build you'll need to move deps on test engine to `dal_test_deps`, but as I suggested make metrics part of common test engine module